### PR TITLE
Add disallowInList option to no-duplicate-selectors

### LINKED ISSUE #3931: Add disallowInList to no-duplicate-selectors
> No-duplicate-selectors doesn't find error, when there're grouping selectors 

> Which rule, if any, is the bug related to?

`no-duplicate-selectors `

> What CSS is needed to reproduce the bug?

```css
.first-selector, .second-selector {
  background: #eee;
}

.second-selector {
  background: #000;
}
```

> What stylelint configuration is needed to reproduce the bug?

e.g.

```json
{
  "rules": {
    "no-duplicate-selectors":  true
  }
}
```

> Which version of stylelint are you using?

`^9.7.0`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

stylelint "**/*.css" --config myconfig.json

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

No

> What did you expect to happen?

Warning to be flagged:

```shell
Unexpected duplicate selector ".second-selector", first used at line 22 (no-duplicate-selectors) [no-duplicate-selectors]
```

> What actually happened (e.g. what warnings or errors did you get)?

There's no warning

### PR DESCRIPTION
**Pull request by devsnice**
> Which issue, if any, is this issue related to?

Closes #3931

### CODE REVIEW

**Comment by devsnice** [comment]
I don't understand why test in ignore.test.js was failed, maybe it's wired with some deps?

**Comment by hudochenkov** [comment]
@devsnice this happens from time to time. We don't know what causes it. I've restarted the job.

File: lib/rules/no-duplicate-selectors/README.md [hunk 1]
```diff
@@ -100,3 +100,22 @@ a {
   & c {}
 }
 ```
+
+## Optional secondary options
+
+### `disallowInList: true | false` (default: `false`)
+
+This option will dissallow duplication selectors within a group selectors
```
Comment (jeddy3) [inline lib/rules/no-duplicate-selectors/README.md]:
```suggestion
This option will also disallow duplicate selectors within selector lists.
```

We like to keep the [vocab](http://apps.workflower.fi/vocabs/css/en) we use consistent. [Selector list](https://www.w3.org/TR/selectors-4/#grouping) comes from the spec.

File: lib/rules/no-duplicate-selectors/README.md [hunk 2]
```diff
@@ -100,3 +100,22 @@ a {
...
+
+For example, with `true`, the following patterns are considered violations:
```
Comment (jeddy3) [inline lib/rules/no-duplicate-selectors/README.md]:
```suggestion
For example, with `true`.

The following patterns are considered violations:
```

It's a bit clunky, but the sentence needs to split for the color highlighting on the website to work, [for example](https://stylelint.io/user-guide/rules/selector-class-pattern/#optional-secondary-options).

**Comment by devsnice** [comment]
> @devsnice Thanks. It's looking good.
> 
> I've made two suggestions to the README.

Thanks! Can you restart job?

**Comment by jeddy3** [comment]
Changelog:

> -   Added: `disallowInList` to `no-duplicate-selectors` ([#3936](https://github.com/stylelint/stylelint/pull/3936)).

**Pull request merged**